lib/widgets/action_sidebar.dart

```diff
@@ -0,0 +1,219 @@
+      left: isExtended ? 0 : 50,
+      right: isExtended ? 0 : -50,
+      top: 0,
+      bottom: 0,
+      child: ConstrainedBox(
+        constraints: const BoxConstraints(maxWidth: 100),
+        child: Stack(
+          children: [
+            Positioned(
+              right: 0,
+              top: 0,
+              bottom: 0,
+              child: ConstrainedBox(
+                constraints: const BoxConstraints(maxWidth: 50),
+                child: Container(
+                  child: Column(
+                    children: widget.actionButtons,
+                  ),
+                  decoration: BoxDecoration(
+                    color: Theme.of(context).bottomAppBarColor.withOpacity(0.9),
+                    boxShadow: [
+                      BoxShadow(
+                        color: Colors.black.withOpacity(0.25),
+                        spreadRadius: 3,
+                        blurRadius: 3,
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+            if (MediaQuery.of(context).size.width < 1100)
+              Positioned(
+                right: 50,
+                child: ActionBarButton(onTap, _controller),
+              )
+            else
+              const SizedBox.shrink(),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class ActionBarButton extends StatelessWidget {
+  const ActionBarButton(this.onTap, this.controller,
+      {this.maxWidth = 50, this.maxHeight = 50});
+
+  final void Function() onTap;
+
+  final AnimationController controller;
+
+  final double maxWidth;
+
+  final double maxHeight;
+
+  Offset get zeroPoint => Offset(maxWidth + 1, 0);
+
+  Set<BezierPoint> getClipPoints() {
+    final BezierPoint first = BezierPoint(
+        maxWidth / 2 - 5, maxHeight / 5, maxWidth / 2 - 5, maxHeight / 2);
+    final BezierPoint second = BezierPoint(
+        maxWidth / 2 - 5, 4 * maxHeight / 5, maxWidth + 1, maxHeight);
+
+    return {first, second};
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ConstrainedBox(
+      constraints: BoxConstraints(maxWidth: maxWidth),
+      child: AspectRatio(
+        aspectRatio: 1,
+        child: GestureDetector(
+          onTap: onTap,
+          child: CustomPaint(
+            painter: BoxShadowPainter(
+              zeroPoint: zeroPoint,
+              points: getClipPoints(),
+            ),
+            child: ClipPath(
+              clipper: CustomMenuClipper(
+                  zeroPoint: zeroPoint, points: getClipPoints()),
+              child: Container(
+                alignment: Alignment.centerRight,
+                color: Theme.of(context).bottomAppBarColor,
+                child: RotationTransition(
+                    turns: Tween(begin: 0.5, end: 0.0).animate(controller),
+                    child: const Icon(Icons.double_arrow)),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class BezierPoint {
```

Comment:
All these classes that aren't needed outside of this file should be private imo.

Reply:
Made private:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/31536835b8d109bd3ef7b078038b0eafbd696c57#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR179-R180